nilearn/decomposition/canica.py

```diff
@@ -216,7 +216,7 @@ def _unmix_components(self, components):
             abs_ica_maps = np.abs(ica_maps)
             threshold = scoreatpercentile(
                 abs_ica_maps,
-                100. - (100. / len(ica_maps)) * ratio)
+                np.maximum(0, 100. - (100. / len(ica_maps)) * ratio))
```

Comment:
How about this: we compute the original formula, which can be negative, and if it is negative, issue a warning, and replace it by zero. This seems the most useful thing to do for users.

Does that make sens?

Reply:
Thank you. alright. Would it perhaps also be more sensible to set the threshold directly to zero instead of the percentile @GaelVaroquaux ?

Reply:
Yes, but it's minor.

The warning message feels important in my eyes, because it improves the usability.